components/script/dom/htmlimageelement.rs

```diff
@@ -186,8 +192,14 @@ impl HTMLImageElement {
 
         match self.current_request.borrow().state {
             // If image's current request's state is broken, then throw an "InvalidStateError" DOMException.
-            State::Broken => Err(Error::InvalidState),
-            State::CompletelyAvailable => Ok(true),
+            State::Broken => {
```

Comment:
I don't think we should do either here, because this is a method that does not transition the state of `current_request`. For example, if `Decode` where to be called, then the earliest point for the promise to resolve would be in the microtask, whereas with these changes an intermediate call to `is_usable` would resolve the promise.